packages/mendel-config/src/transform-config.js

```diff
@@ -1,18 +1,32 @@
 var createValidator = require('./validator');
 var nodeResolveSync = require('resolve').sync;
+var path = require('path');
+
+function _resolve(pluginPath, opt) {
+    try {
+        return nodeResolveSync(pluginPath, opt);
+    } catch (e) {
+        return false;
+    }
+}
+
 
 function TransformConfig(id, transform, {projectRoot}) {
     this.id = id;
+    const basedir = projectRoot;
+    const pluginPackagePath = _resolve(path.join(transform.plugin, 'package.json'), {basedir});
+    const pluginPath = _resolve(transform.plugin, {basedir});
 
-    try {
-        this.plugin = nodeResolveSync(transform.plugin, {basedir: projectRoot});
-        this.mode = require(this.plugin).mode || 'ist';
-    } catch (e) {
-        if (e.code !== 'MODULE_NOT_FOUND') throw e;
-        this.plugin = false;
+    if (pluginPath) {
+        this.mode = [pluginPackagePath, pluginPath]
+            .filter(Boolean)
+            .map(plugin => require(plugin).mode)
+            .filter(Boolean)[0] || 'ist';
```

Comment:
I don't think you are preventing the require. You would prevent if you used `.some`, right?
The map always goes through all resolved. If both plugin and package.json resolves, you require both, no?

Maybe I am missing something.